src/internal/connector/graph/consts.go

```diff
@@ -88,6 +90,14 @@ func (p parallelism) Item() int {
 	return p.item
 }
 
+func (p parallelism) ItemUpload() int {
+	if p.itemUpload == 0 {
```

Comment:
we should probably set an upper bound too. This is going to cause goroutine and buffer creation so we may want to make sure that people don't accidentally set it too high

Reply:
Hmm, I personally don't think we need as this is something that we explicitly mention is about parallelism and it is highly unlikely that people will be assigning random values. But in case people set it to higher value, should we panic or just cap it? Or should we just add tests so to check max?

Reply:
I think we still need some cap on it given that we're making 5MB buffers for each thread we use. It can be a fairly high cap, but we don't want to end up OOMing or dying due to memory pressure if the user accidentally adds an extra 0 or something